hub/core/storage/provider.py

```diff
@@ -0,0 +1,129 @@
+from collections.abc import MutableMapping
+from typing import Optional
+from abc import ABC, abstractmethod
+
+
+class Provider(ABC, MutableMapping):
+    """
+    An abstract base class for implementing a provider.
+    To add a new provider using Provider, create a subclass and implement all 5 abstract methods below.
+    Alternatively, you can inherit from ProviderMapper and have a simpler implementation.
+    """
+
+    @abstractmethod
+    def __getitem__(
+        self,
+        path: str,
+        start_byte: Optional[int] = None,
+        end_byte: Optional[int] = None,
+    ):
+        """
+        Gets the object present at the path.
+        Optionally, if the start_byte and/or end_byte arguments are specified, it only returns required bytes
+
+        Example:
+            local_provider = LocalProvider("/home/ubuntu/Documents/")
+            my_data = local_provider("abc.txt")
```

Comment:
Is this a weird functional notation, or is this supposed to be `[]` and not `()`?